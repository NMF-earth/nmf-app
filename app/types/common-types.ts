export enum GuideCategory {
  kitchen = "kitchen",
  technology = "technology",
  food = "food"
}

export interface Guide {
  title: string;
  category: GuideCategory;
  body: string;
  key: string;
}

