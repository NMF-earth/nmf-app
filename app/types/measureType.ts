export enum MeasureType {
  mass = "mass",
  length = "length",
}
