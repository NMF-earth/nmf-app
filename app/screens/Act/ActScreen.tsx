import React from "react";
import { FlatList } from "react-native";
import { filter, pathEq } from "ramda";

import { ListItem } from "../../components";
import { Guide } from "../../types/guide";
import Guides from "../../../assets/guides/guides.json";
import navigationOptions from "./ActScreen.navigationOptions";
import { navigate } from "../../navigation";
import styles from "./ActScreen.styles";

const getCategory = pathEq(["category"]);

const ActScreen = (props) => {
  const { route, navigation } = props;
  const data = filter(getCategory(route.name), Guides) as Guide[];
  const navigator = navigate(navigation);

  const renderItem = ({ item }) => (
    <ListItem
      key={item.title}
      title={item.title}
      onPress={() => navigator.openActDetails(item)}
    />
  );

  return (
    <FlatList
      style={styles.container}
      data={data}
      renderItem={renderItem}
      keyExtractor={(item) => item.title}
    />
  );
};

ActScreen.navigationOptions = navigationOptions;

export default ActScreen;
