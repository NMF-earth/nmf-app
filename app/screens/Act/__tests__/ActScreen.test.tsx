import React from "react";
import { create } from "react-test-renderer";

import ActScreen from "../ActScreen";
import { GuideCategory } from "../../../types/guide";

const props = {
  route: {
    name: GuideCategory.technology,
  },
  navigation: {
    push: () => {
      // do nothing.
    },
  },
};

it("ActScreen renders correctly", () => {
  const tree = create(<ActScreen {...props} />).toJSON();
  expect(tree).toMatchSnapshot();
});
