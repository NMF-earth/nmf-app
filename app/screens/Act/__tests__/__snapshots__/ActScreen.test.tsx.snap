// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ActScreen renders correctly 1`] = `
<RCTScrollView
  ListFooterComponent={
    <Text.Tertiary
      center={true}
      lightGray={true}
      style={
        Object {
          "paddingTop": 10,
        }
      }
    >
      More coming soon! Stay tuned ✌️
    </Text.Tertiary>
  }
  data={Array []}
  getItem={[Function]}
  getItemCount={[Function]}
  keyExtractor={[Function]}
  onContentSizeChange={[Function]}
  onLayout={[Function]}
  onMomentumScrollBegin={[Function]}
  onMomentumScrollEnd={[Function]}
  onScroll={[Function]}
  onScrollBeginDrag={[Function]}
  onScrollEndDrag={[Function]}
  removeClippedSubviews={false}
  renderItem={[Function]}
  scrollEventThrottle={50}
  stickyHeaderIndices={Array []}
  style={
    Object {
      "backgroundColor": "#FFFFFF",
      "flex": 1,
      "paddingHorizontal": 16,
    }
  }
  viewabilityConfigCallbackPairs={Array []}
>
  <View>
    <View
      onLayout={[Function]}
    >
      <Text.Tertiary
        center={true}
        lightGray={true}
        style={
          Object {
            "paddingTop": 10,
          }
        }
      >
        More coming soon! Stay tuned ✌️
      </Text.Tertiary>
    </View>
  </View>
</RCTScrollView>
`;
